### PR TITLE
[IMP] Fleet Add a parameter to specify the alert delay

### PR DESCRIPTION
**Pull request by jbm-odoo**
Description of the issue/feature this PR addresses:
An alert is sent when the vehicle contract will be outdated in less than 15 days. 

Desired behavior after PR is merged:
There is a parameter than could be set on the res.config to specify the alert delay for outdated contract.

id=1919416

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/fleet/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = ['res.config.settings']
+
+    delay_alert_contract = fields.Integer(string='Delay alert contract outdated', default=30, size=4, config_parameter='hr_fleet.delay_alert_contract')
```
Comment (RomainLibert) [inline addons/fleet/models/res_config_settings.py]: size attribute does nothing here

File: addons/fleet/models/fleet_vehicle.py [hunk 2]
```diff
@@ -115,13 +115,15 @@ def _compute_count_all(self):
 
     @api.depends('log_contracts')
     def _compute_contract_reminder(self):
+        params = self.env['ir.config_parameter'].sudo()
+        delay_alert_contract = params.get_param('hr_fleet.delay_alert_contract', default=30)
```
Comment (RomainLibert) [inline addons/fleet/models/fleet_vehicle.py]: Use int() here and not below

File: addons/fleet/models/fleet_vehicle.py [hunk 3]
```diff
@@ -148,6 +150,8 @@ def _compute_contract_reminder(self):
             record.contract_renewal_name = name
 
     def _search_contract_renewal_due_soon(self, operator, value):
+        params = self.env['ir.config_parameter'].sudo()
+        delay_alert_contract = params.get_param('hr_fleet.delay_alert_contract', default=30)
```
Comment (RomainLibert) [inline addons/fleet/models/fleet_vehicle.py]: Use int() here and not below

File: addons/fleet/models/fleet_vehicle_cost.py [hunk 4]
```diff
@@ -267,9 +267,11 @@ def scheduler_manage_auto_costs(self):
     def scheduler_manage_contract_expiration(self):
         # This method is called by a cron task
         # It manages the state of a contract, possibly by posting a message on the vehicle concerned and updating its status
+        params = self.env['ir.config_parameter'].sudo()
+        delay_alert_contract = params.get_param('hr_fleet.delay_alert_contract', default=30)
```
Comment (RomainLibert) [inline addons/fleet/models/fleet_vehicle_cost.py]: Use int() here and not below

File: addons/fleet/models/fleet_vehicle_cost.py [hunk 5]
```diff
@@ -156,7 +156,7 @@ def _compute_days_left(self):
         otherwise return the number of days before the contract expires
         """
         for record in self:
-            if (record.expiration_date and (record.state == 'open' or record.state == 'expired')):
+            if (record.expiration_date and (record.state == 'open' or record.state == 'diesoon' or record.state == 'expired')):
```
Comment (RomainLibert) [inline addons/fleet/models/fleet_vehicle_cost.py]: You can use `record.state in ['open', 'diesoon', 'expired']`

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**